File: src/test/java/no/nav/syfo/JobbLoggSendOppfolgingsplanTilAltinnTest.java
package no.nav.syfo;

import no.nav.syfo.domain.Oppfolgingsplan;
import no.nav.syfo.metric.Metrikk;
import no.nav.syfo.oppgave.oppfoelgingsdialog.JobbLoggSendOppfoelgingsdialogTilAltinn;
import no.nav.syfo.service.*;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static no.nav.syfo.oppgave.Oppgavetype.OPPFOELGINGSDIALOG_ARKIVER;
import static no.nav.syfo.oppgave.Oppgavetype.OPPFOELGINGSDIALOG_SEND;
import static no.nav.syfo.testhelper.OppfolgingsplanTestUtilsKt.oppfolgingsplanGodkjentTvang;
import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.*;

@RunWith(MockitoJUnitRunner.class)
public class JobbLoggSendOppfolgingsplanTilAltinnTest {

    @Mock
    private OppfolgingsplanService oppfolgingsplanService;
    @Mock
    private JuridiskLoggService juridiskLoggService;
    @Mock
    private Metrikk metrikk;
    @Mock
    private PdfService pdfService;
    @InjectMocks
    private JobbLoggSendOppfoelgingsdialogTilAltinn jobbLoggSendOppfoelgingsdialogTilAltinn;

    @Test
    public void skalUtfoereOppgaveArkiverOppgave() throws Exception {
        assertThat(jobbLoggSendOppfoelgingsdialogTilAltinn.skalUtforeOppgave(OPPFOELGINGSDIALOG_ARKIVER)).isTrue();
    }

    @Test
    public void skalUtfoereOppgaveIkkeArkiverOppgave() throws Exception {
        assertThat(jobbLoggSendOppfoelgingsdialogTilAltinn.skalUtforeOppgave(OPPFOELGINGSDIALOG_SEND)).isFalse();
    }

    @Test
    public void utfoerOppgaveArkiver() throws Exception {
        Oppfolgingsplan oppfolgingsplan = hentOppfoelgingsdialog();
        when(oppfolgingsplanService.hentGodkjentOppfolgingsplan(1L)).thenReturn(oppfolgingsplan);
        when(pdfService.hentPdfTilAltinn(oppfolgingsplan)).thenReturn(getOppfoelgingsdialogPdf());

        jobbLoggSendOppfoelgingsdialogTilAltinn.utfoerOppgave("1");

        verify(juridiskLoggService, times(1)).loggSendOppfoelgingsdialogTilAltinn(any());
    }

    private Oppfolgingsplan hentOppfoelgingsdialog() {
        return oppfolgingsplanGodkjentTvang();
    }

    private byte[] getOppfoelgingsdialogPdf() {
        return new byte[2];
    }
}

File: src/test/kotlin/no/nav/syfo/oppgave/JobbLoggSendOppfolgingsplanTilAltinnTest.kt
package no.nav.syfo.oppgave

import no.nav.syfo.domain.Oppfolgingsplan
import no.nav.syfo.metric.Metrikk
import no.nav.syfo.oppgave.oppfoelgingsdialog.JobbLoggSendOppfoelgingsdialogTilAltinn
import no.nav.syfo.service.*
import no.nav.syfo.testhelper.oppfolgingsplanGodkjentTvang
import org.assertj.core.api.Assertions
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.*
import org.mockito.junit.MockitoJUnitRunner

@RunWith(MockitoJUnitRunner::class)
class JobbLoggSendOppfolgingsplanTilAltinnTest {
    @Mock
    private lateinit var oppfolgingsplanService: OppfolgingsplanService

    @Mock
    private lateinit var juridiskLoggService: JuridiskLoggService

    @Mock
    private lateinit var metrikk: Metrikk

    @Mock
    private lateinit var pdfService: PdfService

    @InjectMocks
    private lateinit var jobbLoggSendOppfoelgingsdialogTilAltinn: JobbLoggSendOppfoelgingsdialogTilAltinn

    @Test
    @Throws(Exception::class)
    fun skalUtfoereOppgaveArkiverOppgave() {
        Assertions.assertThat(jobbLoggSendOppfoelgingsdialogTilAltinn.skalUtforeOppgave(Oppgavetype.OPPFOELGINGSDIALOG_ARKIVER)).isTrue()
    }

    @Test
    @Throws(Exception::class)
    fun skalUtfoereOppgaveIkkeArkiverOppgave() {
        Assertions.assertThat(jobbLoggSendOppfoelgingsdialogTilAltinn.skalUtforeOppgave(Oppgavetype.OPPFOELGINGSDIALOG_SEND)).isFalse()
    }

    @Test
    @Throws(Exception::class)
    fun utfoerOppgaveArkiver() {
        val oppfolgingsplan = hentOppfoelgingsdialog()
        Mockito.`when`(oppfolgingsplanService.hentGodkjentOppfolgingsplan(1L)).thenReturn(oppfolgingsplan)
        Mockito.`when`(pdfService.hentPdfTilAltinn(oppfolgingsplan)).thenReturn(oppfoelgingsdialogPdf)
        jobbLoggSendOppfoelgingsdialogTilAltinn.utfoerOppgave("1")
        Mockito.verify(juridiskLoggService, Mockito.times(1)).loggSendOppfoelgingsdialogTilAltinn(ArgumentMatchers.any())
    }

    private fun hentOppfoelgingsdialog(): Oppfolgingsplan {
        return oppfolgingsplanGodkjentTvang()
    }

    private val oppfoelgingsdialogPdf: ByteArray
        get() = ByteArray(2)
}
